devito/dse/rewriters.py

```diff
@@ -406,6 +411,39 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
         return cluster.rebuild(processed)
 
+    @dse_pass
+    def _skewing(self, cluster, template, **kwargs):
+        """
+        Skew the accesses along the time Dimension.
+        """
+
+        skew_factor = self.DEFAULT_SKEW_FACTOR
+
+        _, sub_iterators, directions = cluster.ispace.args
+
+        skew_dim, mapper, intervals = None, {}, []
+
+        # Check if time dim exists
+        for i in cluster.ispace.intervals:
+            if i.dim.is_Time:
+                skew_dim = i.dim
```

Comment:
I think it's best if you also handle weird/crazy user input , like if there were 2 time dimensions. Pseudocode below

```
skew_dim = {i.dim for i in cluster.ispace.intervals if i.dim.is_Time}
try:
    skew_dim = skew_dim.pop()
except KeyError:
    # No time dimensions -> nothing to do
    return ...
if len(skew_dim) > 0:
    raise ValueError("suitable error msg")
```